pull-upstream-sqlparser.sh

```diff
@@ -0,0 +1,186 @@
+#
+#   1. https://github.com/apache/datafusion-sqlparser-rs
+#
+# Usage: pull-upstream-sqlparser.sh <VERSION>
+
+set -eEu
+
+### Preconditions
+
+if [ -z "${1:-}" ]; then
+  echo "Usage: $(basename "$0") revision-of-sqlparser"
+  echo "   eg. $(basename "$0") v0.53.0"
+  exit 1
+else
+  REV=$1
+fi
+
+PKG_ROOT=$(unset CDPATH; cd "$(dirname "$0")" && pwd)
+if [ ! -d "${PKG_ROOT}/.git" ]; then
+  echo "Please run from the repo root."
+  exit 2
+fi
+
+if command -v gsed >/dev/null; then
+  # Probably on MacOS w/ gnu-sed
+  SED=$(command -v gsed)
+elif [ "$(sed --version | grep -c GNU)" -gt 0 ]; then
+  # Probably on Linux
+  SED=$(command -v sed)
+else
+  echo "Please install GNU's sed ('brew install gnu-sed' on MacOS). No action has been taken."
+  exit 3
+fi
+
+SQLTK_PARSER_RELATIVE_PATH="packages/sqltk-parser"
+SQLTK_PARSER_PATH="${PKG_ROOT}/${SQLTK_PARSER_RELATIVE_PATH}"
+UPSTREAM_REPO_URL="https://github.com/apache/datafusion-sqlparser-rs"
+
+### Helpers
+
+info() {
+  >&2 echo -e "\e[1;34m***\e[0m $*"
+}
+infops() { 
+  # as in, Info PS/postscript. Indents output to show under info() and error() calls
+  >&2 echo -e "    $*"
+}
+error() {
+  >&2 echo -e "\e[1;31m!!!\e[0m $*"
+}
+prompt() {
+  # $1 is the text prompt
+  # $2 is the default value
+  >&2 echo -ne "[\e[1;32m?\e[0m] $1 "
+  read -r PROMPT
+  if [ -z "$PROMPT" ]; then
+    PROMPT=${2:-}
+  fi
+  [ "$(echo "$PROMPT" | grep -ic "y")" -gt 0 ] # returns 0/true or 1/false to caller
+}
+
+### Script
+
+cd "$PKG_ROOT"
+
+#
+# 0. New branch?
+#
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+if [ "$CURRENT_BRANCH" == "main" ]; then
+  if prompt "Current branch seems to be $CURRENT_BRANCH; I recommend creating a separate branch, then re-running this command. Continue anyway?" "n"; then
+    info "Continuing..."
+  else
+    info "Cancelled."
+    exit
+  fi
+fi
+
+#
+# 1. Get the latest from upstream
+#
+if ! git subtree pull --prefix "$SQLTK_PARSER_RELATIVE_PATH" "$UPSTREAM_REPO_URL" "$REV"; then
+  if [ "$(git ls-files --unmerged | wc -l)" -gt 0 ]; then
+    info "Conflicts detected; auto-accepting upstream changes..."
+    infops "NOTE! This means that sqltk-parser changes that conflict with the sqlparser"
+    infops "upstream are automatically being cleared; this will need manual investigation."
+  else
+    error "git subtree pull failed for unknown reason! exiting."
+  fi
+fi
+
+#
+# 2. For all conflicting Rust source files (*.rs) resolve with the upstream version.
+#
+# TODO: This bit is when we'd find out if our added features (eg. LOCK TABLE)
+#       are conflicting with upstream code. It's difficult to make upgrades easy
+#       **and** provide an easy facility for working through these more
+#       selective conflicts. Suggestions very welcome.
+while IFS='' read -r UNMERGED_RS_FILE; do
+  (set -x; git checkout --theirs -- "$UNMERGED_RS_FILE" && git add "$UNMERGED_RS_FILE")
+done < <(git status --porcelain=v1 | grep '^UU .*\.rs$' | "$SED" -e 's/^UU //' )
+echo
```

Comment:
This is still going to suck once `LOCK TABLE` is added to sqltk-parser, I think, but I don't know how to accommodate both easy upgrades *and* easily retaining new features.